app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala

```diff
@@ -21,42 +21,43 @@ import javax.inject.Singleton
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.http.HttpClient
 import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
-import uk.gov.hmrc.tariffclassificationfrontend.models.CaseStatus.CaseStatus
-import uk.gov.hmrc.tariffclassificationfrontend.models.{Case, Queue, Queues, Status => StatusOfTheCase }
-import uk.gov.hmrc.tariffclassificationfrontend.utils.JsonFormatters.{caseFormat, statusFormat}
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.NewEventRequest
+import uk.gov.hmrc.tariffclassificationfrontend.models.{Case, Event, Queue, Queues}
+import uk.gov.hmrc.tariffclassificationfrontend.utils.JsonFormatters.{caseFormat, eventFormat, newEventRequestFormat}
 
 import scala.concurrent.ExecutionContext.Implicits.global
+
 import scala.concurrent.Future
 
 @Singleton
 class BindingTariffClassificationConnector @Inject()(configuration: AppConfig, client: HttpClient) {
 
-  def getOneCase(reference: String)(implicit hc: HeaderCarrier): Future[Option[Case]] = {
+  def findCase(reference: String)(implicit hc: HeaderCarrier): Future[Option[Case]] = {
     val url = s"${configuration.bindingTariffClassificationUrl}/cases/$reference"
     client.GET[Option[Case]](url)
   }
 
-  def getCasesByQueue(queue: Queue)(implicit hc: HeaderCarrier): Future[Seq[Case]] = {
+  def findCasesByQueue(queue: Queue)(implicit hc: HeaderCarrier): Future[Seq[Case]] = {
     val queueId = if (queue == Queues.gateway) "none" else queue.id
     val queryString = s"queue_id=$queueId&assignee_id=none&status=NEW,OPEN,REFERRED,SUSPENDED&sort-by=elapsed-days"
     val url = s"${configuration.bindingTariffClassificationUrl}/cases?$queryString"
     client.GET[Seq[Case]](url)
   }
 
-  def getCasesByAssignee(assignee: String)(implicit hc: HeaderCarrier): Future[Seq[Case]] = {
+  def findCasesByAssignee(assignee: String)(implicit hc: HeaderCarrier): Future[Seq[Case]] = {
     val queryString = s"assignee_id=$assignee&status=NEW,OPEN,REFERRED,SUSPENDED&sort-by=elapsed-days"
     val url = s"${configuration.bindingTariffClassificationUrl}/cases?$queryString"
     client.GET[Seq[Case]](url)
   }
 
-  def updateCaseStatus(caseReference: String, newStatus: CaseStatus)(implicit hc: HeaderCarrier): Future[Case] = {
-    val url = s"${configuration.bindingTariffClassificationUrl}/cases/$caseReference/status"
-    client.PUT[StatusOfTheCase, Case](url, body = StatusOfTheCase(newStatus))
-  }
-
   def updateCase(c: Case)(implicit hc: HeaderCarrier): Future[Case] = {
     val url = s"${configuration.bindingTariffClassificationUrl}/cases/${c.reference}"
     client.PUT[Case, Case](url, body = c)
   }
 
+  def createEvent(c: Case, e: NewEventRequest)(implicit hc: HeaderCarrier): Future[Event] = {
```

Comment:
not a big thing, but you do not need to pass the whole case but just a string (which is the case reference)

Reply:
I'd rather pass around the whole thing to force you to check it exists first